pkg/operator/metrics/metrics.go

```diff
@@ -0,0 +1,62 @@
+package metrics
+
+import (
+	"github.com/blang/semver"
+	"github.com/prometheus/client_golang/prometheus"
+	"k8s.io/component-base/metrics/legacyregistry"
+	"sync"
+	"time"
+)
+
+func NewServiceCAMetrics() *ServiceCAMetrics {
+	s := &ServiceCAMetrics{
+		expiryTime: prometheus.NewGaugeVec(prometheus.GaugeOpts{
+			Name: "service_ca_expiry_time",
+			Help: "Reports the expiry time of service CA used in the cluster.",
+		}, []string{"expiry_time"}),
+		timeLeft: prometheus.NewGaugeVec(prometheus.GaugeOpts{
+			Name: "service_ca_time_left",
+			Help: "Reports the time left before the service CA expires.",
+		}, []string{"time_left"}),
+	}
+	legacyregistry.MustRegister(s)
```

Comment:
constructors with side-effects are bad. Would just fallback to a simple init func here.

Reply:
Thanks for pointing this out. updated the code now.